homeassistant/components/mqtt/__init__.py

```diff
@@ -181,13 +194,13 @@ def mqtt_topic_subscriber(event):
                          event.data[ATTR_PAYLOAD], event.data[ATTR_QOS],
```

Comment:
I was incorrectly under the impression that when you `yield from` that it will execute that task right away before processing the rest of the event loop. Instead it is scheduled on the event loop. With this new found knowledge I can probably do some more clean up 👍